instrumentation/spring/spring-webflux/spring-webflux-5.3/library/src/main/java/io/opentelemetry/instrumentation/spring/webflux/v5_3/SpringWebfluxTelemetryBuilder.java

```diff
@@ -0,0 +1,154 @@
+    clientBuilder.setCapturedClientResponseHeaders(responseHeaders);
+    return this;
+  }
+
+  /**
+   * Sets whether experimental attributes should be set to spans. These attributes may be changed or
+   * removed in the future, so only enable this if you know you do not require attributes filled by
+   * this instrumentation to be stable across versions.
+   */
+  @CanIgnoreReturnValue
+  public SpringWebfluxTelemetryBuilder setCaptureExperimentalSpanAttributes(
+      boolean captureExperimentalSpanAttributes) {
+    clientBuilder.setCaptureExperimentalSpanAttributes(captureExperimentalSpanAttributes);
+    return this;
+  }
+
+  /**
+   * Adds an additional {@link AttributesExtractor} to invoke to set attributes to instrumented
+   * items.
+   */
+  @CanIgnoreReturnValue
+  public SpringWebfluxTelemetryBuilder addServerAttributesExtractor(
+      AttributesExtractor<ServerWebExchange, ServerWebExchange> attributesExtractor) {
+    serverAdditionalExtractors.add(attributesExtractor);
+    return this;
+  }
+
+  /**
+   * Configures the HTTP request headers that will be captured as span attributes from server
+   * instrumentation.
+   *
+   * @param requestHeaders A list of HTTP header names.
+   */
+  @CanIgnoreReturnValue
+  public SpringWebfluxTelemetryBuilder setCapturedServerRequestHeaders(
+      List<String> requestHeaders) {
+    httpServerAttributesExtractorBuilder.setCapturedRequestHeaders(requestHeaders);
+    return this;
+  }
+
+  /**
+   * Configures the HTTP response headers that will be captured as span attributes from server
+   * instrumentation.
+   *
+   * @param responseHeaders A list of HTTP header names.
+   */
+  @CanIgnoreReturnValue
+  public SpringWebfluxTelemetryBuilder setCapturedServerResponseHeaders(
+      List<String> responseHeaders) {
+    httpServerAttributesExtractorBuilder.setCapturedResponseHeaders(responseHeaders);
+    return this;
+  }
+
+  /**
+   * Returns a new {@link SpringWebfluxTelemetry} with the settings of this {@link
+   * SpringWebfluxTelemetryBuilder}.
+   */
+  public SpringWebfluxTelemetry build() {
+
+    // headers are injected elsewhere; ClientRequest is immutable
+    Instrumenter<ClientRequest, ClientResponse> clientInstrumenter = clientBuilder.build();
+
+    WebfluxServerHttpAttributesGetter serverAttributesGetter =
+        WebfluxServerHttpAttributesGetter.INSTANCE;
+    SpanNameExtractor<ServerWebExchange> serverSpanNameExtractor =
+        HttpSpanNameExtractor.create(serverAttributesGetter);
+
+    Instrumenter<ServerWebExchange, ServerWebExchange> serverInstrumenter =
+        Instrumenter.<ServerWebExchange, ServerWebExchange>builder(
+                openTelemetry, INSTRUMENTATION_NAME, serverSpanNameExtractor)
+            .setSpanStatusExtractor(HttpSpanStatusExtractor.create(serverAttributesGetter))
+            .addAttributesExtractor(httpServerAttributesExtractorBuilder.build())
+            .addAttributesExtractors(serverAdditionalExtractors)
```

Comment:
Can you add metrics?
```suggestion
            .addAttributesExtractors(serverAdditionalExtractors)
            .addOperationMetrics(HttpServerMetrics.get())
```